physics/SFC_Models/Land/Noahmp/lnd_iau_mod.F90

```diff
@@ -0,0 +1,879 @@
+      integer              :: me              !< MPI rank designator
+      integer              :: mpi_root          !< MPI rank of master atmosphere processor
+      character(len=64)    :: fn_nml          !< namelist filename for surface data cycling
+      real(kind=kind_phys) :: dtp             !< physics timestep in seconds
+      real(kind=kind_phys) :: fhour           !< current forecast hour
+!      character(len=:), pointer, dimension(:) :: input_nml_file => null() !<character string containing full namelist
+                                                                          !< for use with internal file reads
+!      integer              :: input_nml_file_length    !<length (number of lines) in namelist for internal reads
+
+      integer              :: ntimes
+
+  end type land_iau_control_type
+
+  type(land_iau_state_type) :: Land_IAU_state
+  public land_iau_control_type, land_iau_external_data_type, land_iau_mod_set_control, &
+         land_iau_mod_init, land_iau_mod_getiauforcing, land_iau_mod_finalize, calculate_landinc_mask
+
+contains
+
+subroutine land_iau_mod_set_control(Land_IAU_Control,fn_nml,input_nml_file_i, me, mpi_root, &
+                                   isc, jsc, nx, ny, tile_num, nblks, blksz, &
+                                   lsoil, lsnow_lsm, dtp, fhour, errmsg, errflg)          !nlunit
+
+   type (land_iau_control_type), intent(inout) :: Land_IAU_Control
+   character(*), intent(in)                    :: fn_nml          !< namelist filename for surface data cycling
+   character(len=:), intent(in), dimension(:), pointer :: input_nml_file_i
+   integer, intent(in)                        :: me, mpi_root          !< MPI rank of master atmosphere processor   
+   integer, intent(in)                        :: isc, jsc, nx, ny, tile_num, nblks, lsoil, lsnow_lsm
+   integer, dimension(:),          intent(in) :: blksz   !(one:) !GFS_Control%blksz
+   real(kind=kind_phys), intent(in)           :: dtp             !< physics timestep in seconds
+   real(kind=kind_phys), intent(in)           :: fhour           !< current forecast hour
+   character(len=*),              intent(out) :: errmsg
+   integer,                       intent(out) :: errflg
+   
+   integer                                    :: nb, ix
+   integer                                    :: nlunit = 360          ! unit for namelist  !, intent(in)
+   integer                                    :: ios
+   logical                                    :: exists
+   character(len=512)                         :: ioerrmsg
+   !character(len=32)                          :: fn_nml = "input.nml"
+   character(len=:), pointer, dimension(:)    :: input_nml_file => null()
+   integer                                    :: input_nml_file_length    !< length(number of lines) in namelist for internal reads
+   character(len=4)                           :: iosstr
+
+   !> these are not available through the CCPP interface so need to read them from namelist file

```

Comment:
If it is preferrable that  these variables are included in the existing `gfs_physics_nml` and passed in through the CCPP (like other iau namelist variables), then we can do that too. There's nothing wrong with reading in nml from a CCPP scheme in the init phase, but if we want to be consistent with past iau code, we can change it. Let me know if you would rather do that and whether you need help with it.

Reply:
@grantfirl I now see that separate nml for land iau would be better, so no need to include in gfs_physics nml. I will get rid of the old comment.